src/engine/bank05.asm

```diff
@@ -54,31 +54,1059 @@ PointerTable_14000: ; 14000 (05:4000)
+	or a
+	jr nz, .evolution_card
+	ld a, DUELVARS_NUMBER_OF_POKEMON_IN_PLAY_AREA
+	call GetTurnDuelistVariable
+	cp MAX_PLAY_AREA_POKEMON
+	ccf
+	ret
+
+.evolution_card
+	bank1call IsPrehistoricPowerActive
+	ret c
+	ld a, DUELVARS_NUMBER_OF_POKEMON_IN_PLAY_AREA
+	call GetTurnDuelistVariable
+	ld c, a
+	ld b, 0
+.loop
+	push bc
+	ld e, b
+	ldh a, [hTempCardIndex_ff9f]
+	ld d, a
+	call CheckIfCanEvolveInto
+	pop bc
+	ret nc
+	inc b
+	dec c
+	jr nz, .loop
+	scf
+	ret
+
+.trainer_card
+	bank1call CheckCantUseTrainerDueToHeadache
+	ret c
+	call LoadNonPokemonCardEffectCommands
+	ld a, EFFECTCMDTYPE_INITIAL_EFFECT_1
+	call TryExecuteEffectCommandFunction
+	ret
+; 0x1438c
+
+; loads all the energy cards
+; in hand in wDuelTempList
+; return carry if no energy cards found
+CreateEnergyCardListFromHand: ; 1438c (5:438c)
+	push hl
+	push de
+	push bc
+	ld de, wDuelTempList
+	ld b, a
+	ld a, DUELVARS_NUMBER_OF_CARDS_IN_HAND
+	call GetTurnDuelistVariable
+	ld c, a
+	inc c
+	ld l, LOW(wOpponentHand)
+	jr .decrease
+
+.loop
+	ld a, [hli]
+	push de
+	call GetCardIDFromDeckIndex
+	call GetCardType
+	pop de
+	and TYPE_ENERGY
+	jr z, .decrease
+	dec hl
+	ld a, [hli]
+	ld [de], a
+	inc de
+.decrease
+	dec c
+	jr nz, .loop
+
+	ld a, $ff
+	ld [de], a
+	pop bc
+	pop de
+	pop hl
+	ld a, [wDuelTempList]
+	cp $ff
+	ccf
+	ret
+; 0x143bf
+
+; looks for card ID in hand and
+; sets carry if a card wasn't found
+; as opposed to LookForCardIDInHandList
+; this function doesn't create a list
+; and preserves hl, de and bc
+; input:
+;	a = card ID
+; output:
+; 	a = card deck index, if found
+;	carry set if NOT found
+LookForCardIDInHand: ; 143bf (5:43bf)
+	push hl
+	push de
+	push bc
+	ld b, a
+	ld a, DUELVARS_NUMBER_OF_CARDS_IN_HAND
+	call GetTurnDuelistVariable
+	ld c, a
+	inc c
+	ld l, DUELVARS_HAND
+	jr .next
+
+.loop
+	ld a, [hli]
+	call GetCardIDFromDeckIndex
+	ld a, e
+	cp b
+	jr z, .no_carry
+.next
+	dec c
+	jr nz, .loop
+
+	pop bc
+	pop de
+	pop hl
+	scf
+	ret
+
+.no_carry
+	dec hl
+	ld a, [hl]
+	pop bc
+	pop de
+	pop hl
+	or a
+	ret
+; 0x143e5
+
+; stores in wDamage, wAIMinDamage and wAIMaxDamage the calculated damage
+; done to the defending Pokémon by a given card and move
+; input:
+; 	a = move index to take into account
+; 	[hTempPlayAreaLocation_ff9d] = location of attacking card to consider
+CalculateMoveDamage_VersusDefendingCard: ; 143e5 (5:43e5)
+	ld [wSelectedMoveIndex], a
+	ld e, a
+	ldh a, [hTempPlayAreaLocation_ff9d]
+	add DUELVARS_ARENA_CARD
+	call GetTurnDuelistVariable
+	ld d, a
+	call CopyMoveDataAndDamage_FromDeckIndex
+	ld a, [wLoadedMoveCategory]
+	cp POKEMON_POWER
+	jr nz, .is_move
+
+; is a Pokémon Power
+; set wDamage, wAIMinDamage and wAIMaxDamage to zero
+	ld hl, wDamage
+	xor a
+	ld [hli], a
+	ld [hl], a
+	ld [wAIMinDamage], a
+	ld [wAIMaxDamage], a
+	ld e, a
+	ld d, a
+	ret
+
+.is_move
+; set wAIMinDamage and wAIMaxDamage to damage of move
+	ld a, [wDamage]
+	ld [wAIMinDamage], a
+	ld [wAIMaxDamage], a
+	ld a, EFFECTCMDTYPE_AI
```

Comment:
Many effects use this command to set specific values to ``wAIMinDamage`` and ``wAIMaxDamage``, like to account for poison, multiple hits, etc. It could be appropriate to mention it as from your L690 comment it could be easily overlooked.